### ISSUE TITLE
Python 3.11 BaseExceptionGroup mistyped as ExceptionGroup

### ISSUE THREAD

**Issue by VermiIIi0n** [microsoft/pylance-release#4183]
## Environment data

-   Language Server version: 2023.3.40
-   OS and version: MacOS Ventura
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11

## Code Snippet

```python
try:
    raise BaseException()
except* BaseException as eg:
    reveal_type(eg)
```

## Repro Steps

Copy the snippet to VS Code

## Expected behavior

Revealed type should be `BaseExceptionGroup[BaseException]`

## Actual behavior

Pylance reported type `ExceptionGroup[BaseException]` which is wrong and impossible. 

<img width="589" alt="image" src="https://user-images.githubusercontent.com/104612478/229339367-703d5b85-d6db-4510-b8e6-0da50b5758c6.png">

Should be an easy fix 🙂.

## Logs

```
XXX
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next version of pyright and a future version of pylance.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.4.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023411-5-April-2023-PreRelease)
